jetty-websocket/websocket-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -86,21 +85,24 @@
     private boolean stopAtShutdown = true;
 
     /**
-     * Instantiate a WebSocketClient with defaults
+     * Instantiate a WebSocketClient with defaults.
      */
     public WebSocketClient()
     {
-        this((HttpClient)null);
+        this(new HttpContainerScope(HttpClientProvider.get(null)));
+        // Add as bean, as HttpClient was created in this class
+        addBean(this.httpClient);
```

Comment:
This call to add the `HttpClient` as a bean is repeated in every constructor, while it should only be in the newly introduced constructor.

Reply:
Nope, that's incorrect, there are 2 paths that have external clients.
Can't do that in the common constructor.